sphinx/builders/linkcheck.py

```diff
@@ -437,9 +438,10 @@ def _check_uri(self, uri: str, hyperlink: Hyperlink) -> tuple[str, str, int]:
             except HTTPError as err:
                 error_message = str(err)
 
-                # Unauthorised: the reference probably exists
+                # Unauthorized: the client did not provide required credentials
                 if status_code == 401:
-                    return 'working', 'unauthorized', 0
+                    status = 'working' if self.allow_unauthorized else 'broken'
```

Comment:
I suppose we'll also want a deprecation warning when the `linkcheck_allow_unauthorized` option is configured to `False`, to indicate that it will be removed.

I'm pausing work at the moment, but I'll look into that within the next day or two.

Reply:
> I suppose we'll also want a deprecation warning when the `linkcheck_allow_unauthorized` option is configured to `False`, to indicate that it will be removed.

Self-correction (implemented): if we're removing the setting, we'll want to issue a warning if _any_ value is configured for it.